src/Compatibility/ApiDiff/Microsoft.DotNet.ApiDiff.Tool/Program.cs

```diff
@@ -0,0 +1,209 @@
+/// different versions of the same assembly, using the specified command line options.
+/// </summary>
+public static class Program
+{
+    public static async Task Main(string[] args)
+    {
+        RootCommand rootCommand = new("genapidiff");
+
+        Option<string> optionBeforeAssembliesFolderPath = new(["--before", "-b"])
+        {
+            Description = "The path to the folder containing the old (before) assemblies.",
+            Arity = ArgumentArity.ExactlyOne,
+            IsRequired = true
+        };
+
+        Option<string> optionBeforeRefAssembliesFolderPath = new(["--refbefore", "-rb"])
+        {
+            Description = "The path to the folder containing the old (before) reference assemblies.",
+            Arity = ArgumentArity.ExactlyOne,
+            IsRequired = false
+        };
+
+        Option<string> optionAfterAssembliesFolderPath = new(["--after", "-a"])
+        {
+            Description = "The path to the folder containing the new (after) assemblies.",
+            Arity = ArgumentArity.ExactlyOne,
+            IsRequired = false
+        };
+
+        Option<string> optionAfterRefAssembliesFolderPath = new(["--refafter", "-ra"])
+        {
+            Description = "The path to the folder containing the new (after) reference assemblies.",
+            Arity = ArgumentArity.ExactlyOne,
+            IsRequired = false
+        };
+
+        Option<string> optionOutputFolderPath = new(["--output", "-o"])
+        {
+            Description = "The path to the output folder.",
+            Arity = ArgumentArity.ExactlyOne,
+            IsRequired = true
+        };
+
+        Option<string> optionBeforeFriendlyName = new(["--beforeFriendlyName", "-bfn"])
+        {
+            Description = "The friendly name to describe the 'before' assembly.",
+            Arity = ArgumentArity.ExactlyOne,
+            IsRequired = true
+        };
+
+        Option<string> optionAfterFriendlyName = new(["--afterFriendlyName", "-afn"])
+        {
+            Description = "The friendly name to describe the 'after' assembly.",
+            Arity = ArgumentArity.ExactlyOne,
+            IsRequired = true
+        };
+
+        Option<string> optionTableOfContentsTitle = new(["--tableOfContentsTitle", "-tc"])
+        {
+            Description = "The title of the markdown table of contents file that is placed in the output folder.",
+            Arity = ArgumentArity.ExactlyOne,
+            IsRequired = true
+        };
+
+        Option<string[]?> optionAssembliesToExclude = new(["--assembliesToExclude", "-eas"], () => null)
+        {
+            Description = "Assemblies to exclude from the diff.",
+            Arity = ArgumentArity.ZeroOrMore,
+            IsRequired = false,
+        };
+
+        Option<string[]?> optionAttributesToExclude = new(["--attributesToExclude", "-eattrs"], () => null)
+        {
+            Description = "Attributes to exclude from the diff.",
```

Comment:
What's the format?  Is it a file with docIDs?  If not - can we put docIDs on the commandline safely?  They have things like `<` and `>` etc that might get sketchy.

Reply:
I'll change this to instead receive a file path. The contents will then still be passed as an array of strings so this can easily be unit tested without involving files.

I'll also mention the format.